Dockerfile

```diff
@@ -15,7 +15,11 @@ RUN mkdir -p /tmp/python && \
 COPY pyproject.toml /tmp/python
 RUN pip install --no-cache-dir /tmp/python/.
 
-# User corresponds to dedicated shiny-server user on server. Necessary for
-# proper write permissions to target directories
-USER 1003:0
+# User corresponds to dedicated shiny-server user on server OR the user
+# specified via build args. Necessary for proper write permissions to target
+# directories. Note that the group must remain root due to issues with the
+# upstream container
+ARG USER_ID=1003
+
+USER ${USER_ID}:0
```

Comment:
Whoever is hacking on this repo will need to run `docker compose build` to create a version of the container that users their own UID. This will let them remove/edit files created by Spark in the `target/` directory (which would otherwise be owned by `shiny-server` or `root`).